src/AppBundle/Repository/Repository.php

```diff
@@ -318,6 +320,25 @@ public function getUsernameFromId(int $userId): ?string
         return $ret[0]['user_name'] ?? null;
     }
 
+    /**
+     * Get actor IDs for the given usernames.
+     * @param string $dbName
+     * @param string[] $usernames
+     * @return int[]
+     */
+    public function getActorIdsFromUsernames(string $dbName, array $usernames): array
+    {
+        $conn = $this->getReplicaConnection();
+        $rqb = $conn->createQueryBuilder();
+
+        $rqb->select('actor_id')
+            ->from("$dbName.actor")
+            ->where('actor_name IN (:usernames)')
+            ->setParameter('usernames', $usernames, Connection::PARAM_STR_ARRAY);
+
+        return $this->executeQueryBuilder($rqb)->fetchAll(PDO::FETCH_FUNC, 'intval');
```

Comment:
`ResultStatement::fetchAll()` seems to want type `int|null` as the second argument.

Reply:
It documents the FETCH_FUNC usage, so the type annotation is wrong. In any case, there's a test that verify that it works as intended.